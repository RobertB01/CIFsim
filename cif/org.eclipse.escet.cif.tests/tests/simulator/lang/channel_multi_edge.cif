//////////////////////////////////////////////////////////////////////////////
// Copyright (c) 2010, 2022 Contributors to the Eclipse Foundation
//
// See the NOTICE file(s) distributed with this work for additional
// information regarding copyright ownership.
//
// This program and the accompanying materials are made available
// under the terms of the MIT License which is available at
// https://opensource.org/licenses/MIT
//
// SPDX-License-Identifier: MIT
//////////////////////////////////////////////////////////////////////////////

event e;
event void ev1;
event void ev2;
event int ei1;
event int ei2;

automaton as:
  location l1:
    initial;
    urgent;
    edge ev1!, ev2!, ei1!1, ei2!2, e;
end

automaton ae:
  location l1:
    initial;
    edge e goto l2;

  location l2;
end

automaton arv1:
  location l1:
    initial;
    edge ev1? goto l2;

  location l2;
end

automaton arv2:
  location l1:
    initial;
    edge ev2? goto l2;

  location l2;
end

automaton ari1:
  disc int x;

  location l1:
    initial;
    edge ei1? do x := ? goto l2;

  location l2;
end

automaton ari2:
  disc int x;

  location l1:
    initial;
    edge ei2? do x := ? goto l2;

  location l2;
end
