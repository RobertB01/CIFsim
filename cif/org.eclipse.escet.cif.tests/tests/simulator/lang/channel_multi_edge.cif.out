Initial state: time=0.0, ae=l1, ari1=l1, ari1.x=0, ari2=l1, ari2.x=0, arv1=l1, arv2=l1, as=l1

Possible transitions:
  #1: event e
  #2: event ei1 (communicates value 1)
  #3: event ei2 (communicates value 2)
  #4: event ev1 (communicates no value)
  #5: event ev2 (communicates no value)

Transition: event e
State: time=0.0, ae=l2, ari1=l1, ari1.x=0, ari2=l1, ari2.x=0, arv1=l1, arv2=l1, as=l1

Possible transitions:
  #1: event ei1 (communicates value 1)
  #2: event ei2 (communicates value 2)
  #3: event ev1 (communicates no value)
  #4: event ev2 (communicates no value)

Transition: event ei1 (communicates value 1)
State: time=0.0, ae=l2, ari1=l2, ari1.x=1, ari2=l1, ari2.x=0, arv1=l1, arv2=l1, as=l1

Possible transitions:
  #1: event ei2 (communicates value 2)
  #2: event ev1 (communicates no value)
  #3: event ev2 (communicates no value)

Transition: event ei2 (communicates value 2)
State: time=0.0, ae=l2, ari1=l2, ari1.x=1, ari2=l2, ari2.x=2, arv1=l1, arv2=l1, as=l1

Possible transitions:
  #1: event ev1 (communicates no value)
  #2: event ev2 (communicates no value)

Transition: event ev1 (communicates no value)
State: time=0.0, ae=l2, ari1=l2, ari1.x=1, ari2=l2, ari2.x=2, arv1=l2, arv2=l1, as=l1

Transition: event ev2 (communicates no value)
State: time=0.0, ae=l2, ari1=l2, ari1.x=1, ari2=l2, ari2.x=2, arv1=l2, arv2=l2, as=l1

Simulation resulted in deadlock.
